From 8a8f12059cc4dc4fd5c3719dece457bc8e442e32 Mon Sep 17 00:00:00 2001
From: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
Date: Thu, 1 Jul 2021 18:20:10 +0300
Subject: [PATCH 1/6] Makefile.inc: fix depending on $(SBSIGNTOOLS)

The variable points to a directory, which might have its timestamp newer
for reasons unrelated to the build.  This might lead to signing failing
because the tools just weren't built.  Depend on a particular build artifact instead to fix the issue.

Change-Id: Ic0d47709eb447fd8d29cb41242280e336740ea52
Signed-off-by: Sergii Dmytruk <sergii.dmytruk@3mdeb.com>
---
 Makefile.inc | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile.inc b/Makefile.inc
index 0bce7b69c0..0ae6eb6bcf 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -586,7 +586,7 @@ $(ECCTOOL):
 	cd $(objutil)/ffs && autoreconf -i && ./configure
 	+$(MAKE) -C $(objutil)/ffs
 
-$(SBSIGNTOOLS):
+$(SBSIGNTOOLS)/create-container:
 	@printf "    Compile SB SIGNING UTILS\n"
 	cp -r $(top)/3rdparty/sb-signing-utils $(objutil)
 	cd $(objutil)/sb-signing-utils && autoreconf -i -Wno-unsupported && ./configure
@@ -703,7 +703,7 @@ install-git-commit-clangfmt:
 include util/crossgcc/Makefile.inc
 
 .PHONY: tools
-tools: $(objutil)/kconfig/conf $(objutil)/kconfig/toada $(CBFSTOOL) $(objutil)/cbfstool/cbfs-compression-tool $(FMAPTOOL) $(RMODTOOL) $(IFWITOOL) $(objutil)/nvramtool/nvramtool $(objutil)/sconfig/sconfig $(IFDTOOL) $(CBOOTIMAGE) $(AMDFWTOOL) $(AMDCOMPRESS) $(FUTILITY) $(BINCFG) $(IFITTOOL) $(objutil)/supermicro/smcbiosinfo $(ECCTOOL) $(SBSIGNTOOLS)
+tools: $(objutil)/kconfig/conf $(objutil)/kconfig/toada $(CBFSTOOL) $(objutil)/cbfstool/cbfs-compression-tool $(FMAPTOOL) $(RMODTOOL) $(IFWITOOL) $(objutil)/nvramtool/nvramtool $(objutil)/sconfig/sconfig $(IFDTOOL) $(CBOOTIMAGE) $(AMDFWTOOL) $(AMDCOMPRESS) $(FUTILITY) $(BINCFG) $(IFITTOOL) $(objutil)/supermicro/smcbiosinfo $(ECCTOOL) $(SBSIGNTOOLS)/create-container
 
 ###########################################################################
 # Common recipes for all stages
@@ -1162,7 +1162,7 @@ add_intermediate = \
 	$(1): $(obj)/coreboot.pre $(2) | $(INTERMEDIATE) \
 	$(eval INTERMEDIATE+=$(1)) $(eval PHONY+=$(1))
 
-$(obj)/coreboot.rom: $(obj)/coreboot.pre $(RAMSTAGE) $(CBFSTOOL) $(ECCTOOL) $(SBSIGNTOOLS) $$(INTERMEDIATE)
+$(obj)/coreboot.rom: $(obj)/coreboot.pre $(RAMSTAGE) $(CBFSTOOL) $(ECCTOOL) $(SBSIGNTOOLS)/create-container $$(INTERMEDIATE)
 	@printf "    CBFS       $(subst $(obj)/,,$(@))\n"
 # The full ROM may be larger than the CBFS part, so create an empty
 # file (filled with \377 = 0xff) and copy the CBFS image over it.
-- 
2.17.6

